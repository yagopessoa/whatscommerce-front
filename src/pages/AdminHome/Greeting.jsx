import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

import Card from '../../components/Card/Card';
import { Container } from '../../components/CommonStyles/CommonStyles';
import { CardContent } from './AdminHome.styles';
import { UserActions } from '../../store/ducks/user';

const Greeting = (props = { onConfirm: () => {} }) => {
  const dispatch = useDispatch();
  const { logout } = UserActions;

  const { name } = useSelector(({ user }) => user);
  const firstName = () => (name ? name.split(' ')[0] : '');

  const { onConfirm } = props;

  return (
    <Container>
      <Card>
        <CardContent>
          <b data-testid="hello-msg">Bem-vindo(a), {firstName()}</b>
          <br />
          <br />
          Agora é hora de prepararmos sua lojinha para seus clientes, vamos lá?
          <br />
          <br />
          <button type="button" onClick={onConfirm} data-testid="create-store-btn">
            Criar minha lojinha
          </button>
        </CardContent>
      </Card>
      <br />
      <br />
      <button type="button" onClick={() => dispatch(logout())}>
        Logout
      </button>
    </Container>
  );
};

export default Greeting;
