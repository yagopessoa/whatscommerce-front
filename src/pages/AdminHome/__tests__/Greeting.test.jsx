import React from 'react';
import sinon from 'sinon';
import { render, fireEvent } from '../../../test-utils';

import Greeting from '../Greeting';

describe('Greeting tests', () => {
  const props = { onConfirm: sinon.spy() };
  const initialState = { user: { name: 'UserMockedName' } };

  let wrapper;

  beforeEach(() => {
    wrapper = render(<Greeting {...props} />, {
      initialState,
    });
  });

  it('should render correctly', () => {
    const { container } = wrapper;
    expect(container).toMatchSnapshot();
  });

  it('should call onClose on cancel or exit click + start with confirm button disabled', async () => {
    const { findByTestId } = wrapper;

    const helloNode = await findByTestId('hello-msg');
    const createButtonNode = await findByTestId('create-store-btn');
    const { onConfirm } = props;

    expect(helloNode.textContent).toContain('UserMockedName');

    fireEvent.click(createButtonNode);
    expect(onConfirm.called).toBeTruthy();
  });
});
