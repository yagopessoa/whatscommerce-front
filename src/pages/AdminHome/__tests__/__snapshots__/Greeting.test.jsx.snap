// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting tests should render correctly 1`] = `
<div>
  <div
    class="sc-AxirZ hOjuvP"
  >
    <div
      class="sc-Axmtr iVYyva"
    >
      <div
        class="sc-AxmLO kXDhGw"
      >
        <b
          data-testid="hello-msg"
        >
          Bem-vindo(a), 
          UserMockedName
        </b>
        <br />
        <br />
        Agora é hora de prepararmos sua lojinha para seus clientes, vamos lá?
        <br />
        <br />
        <button
          data-testid="create-store-btn"
          type="button"
        >
          Criar minha lojinha
        </button>
      </div>
    </div>
    <br />
    <br />
    <button
      type="button"
    >
      Logout
    </button>
  </div>
</div>
`;
